src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -38,13 +38,29 @@ internal sealed class InMethodBinder : LocalScopeBinder
 #endif
 
         public InMethodBinder(MethodSymbol owner, Binder enclosing)
-            : base(enclosing, enclosing.Flags & ~BinderFlags.AllClearedAtExecutableCodeBoundary)
+            : base(enclosing, ConstructFlags(owner, enclosing))
         {
             Debug.Assert(!enclosing.Flags.Includes(BinderFlags.InCatchFilter));
             Debug.Assert((object)owner != null);
             _methodSymbol = owner;
         }
 
+        private static BinderFlags ConstructFlags(MethodSymbol owner, Binder enclosing)
```

Comment:
>ConstructFlags [](http://example.com/codeflow?start=35&length=14)

Consider converting this to a local function. It looks like it is used only in one place.